include/h2o.h

```diff
@@ -1006,8 +989,26 @@ struct st_h2o_conn_t {
         char str[H2O_UUID_STR_RFC4122_LEN + 1];
         uint8_t is_initialized;
     } _uuid;
+    /* internal structure */
+    h2o_linklist_t _conns;
 };
 
+#define H2O_CONN_LIST_FOREACH(decl_var, conn_list, block)                                                                          \
+    do {                                                                                                                           \
+        size_t conn_list_len = sizeof(conn_list) / sizeof((conn_list)[0]);                                                         \
+        h2o_linklist_t **_conn_list = (conn_list);                                                                                 \
+        for (size_t i = 0; i < conn_list_len; i++) {                                                                               \
+            for (h2o_linklist_t *_node = _conn_list[i]->next, *_node_next; _node != _conn_list[i]; _node = _node_next) {           \
+                _node_next = _node->next;                                                                                          \
+                h2o_conn_t *_h2o_conn = H2O_STRUCT_FROM_MEMBER(h2o_conn_t, _conns, _node);                                         \
+                decl_var = (void *)_h2o_conn;                                                                                      \
+                {                                                                                                                  \
+                    block                                                                                                          \
+                }                                                                                                                  \
+            }                                                                                                                      \
+        }                                                                                                                          \
+    } while (0)
```

Comment:
I know that this macro was in the original WIP work, but honestly speaking, I'm not sure if this abstraction improves maintainability or not. WDYT?

Reply:
Follow up here: https://github.com/h2o/h2o/pull/2932#discussion_r798749534